src/components/TransactionHistoryRows/TransactionDetails/ApprovalAmount.tsx

```diff
@@ -0,0 +1,63 @@
+import { MaxUint256 } from '@ethersproject/constants'
+import { AssetId } from '@shapeshiftoss/caip'
+import { TxMetadata } from '@shapeshiftoss/chain-adapters'
+import { useMemo } from 'react'
+import { useTranslate } from 'react-polyglot'
+import { bn, bnOrZero } from 'lib/bignumber/bignumber'
+import { selectAssetById, selectMarketDataById } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+import { Row } from './Row'
+import { Text } from './Text'
+
+export const ApprovalAmount = ({
+  assetId,
+  value,
+  isRevoke,
+  parser,
+}: {
+  assetId: AssetId
+  value: string
+  isRevoke: boolean
+  parser: TxMetadata['parser'] | undefined
+}) => {
+  const translate = useTranslate()
+
+  const approvedAsset = useAppSelector(state => selectAssetById(state, assetId))
+  const approvedAmount = bn(value).div(bn(10).pow(approvedAsset.precision)).toString()
+  const approvedAssetMarketData = useAppSelector(state => selectMarketDataById(state, assetId))
+
+  const amountOrDefault = useMemo(() => {
+    if (isRevoke) return translate(`transactionRow.parser.${parser}.revoke`)
+    // If equal to max. Solidity uint256 value or greater than/equal to max supply, we can infer infinite approvals without market data
+    if (
+      (approvedAssetMarketData.maxSupply &&
+        bn(approvedAssetMarketData.maxSupply).gte(0) &&
+        bn(approvedAmount).gte(approvedAssetMarketData.maxSupply)) ||
+      bn(value).isEqualTo(MaxUint256.toString())
+    )
+      return translate(`transactionRow.parser.${parser}.infinite`)
+    // We don't have market data for that asset thus can't know whether or not it's infinite
+    if (bnOrZero(approvedAssetMarketData.supply).isZero()) return approvedAmount
+    // We have market data and the approval is greater than it, so we can assume it's infinite
+    if (bn(approvedAmount).gte(approvedAssetMarketData.supply ?? '0'))
```

Comment:
This might be unsafe - it could be true at the time of the approval and false later on, might want to use historical market data?

Reply:
To me it makes sense to see the current state - if I approved "infinite" at the time, but it no longer is, I'd expect to see the approved amount - so this is good as is 👌 

The reverse is a little jarring, though - it was a fixed amount, and is now "infinite"...